packages/devtools_app/lib/src/timeline/flutter/flutter_frames_chart.dart

```diff
@@ -99,269 +78,402 @@ class _FlutterFramesChartState extends State<FlutterFramesChart>
+            themeData: themeData,
+          ),
+        );
+        return Stack(
+          children: [
+            chartAxisPainter,
+            Padding(
+              padding: const EdgeInsets.only(left: yAxisUnitsSpace),
+              child: chart,
+            ),
+            fpsLinePainter,
+          ],
+        );
       },
-      drawGridBackground: false,
-//      dragXEnabled: true,
-//      dragYEnabled: true,
-//      scaleXEnabled: true,
-//      scaleYEnabled: true,
-//      pinchZoomEnabled: false,
-//      maxVisibleCount: 60,
-      drawBarShadow: false,
-      description: desc,
-      highLightPerTapEnabled: true,
-      marker: SelectedDataPoint(onSelected: onBarSelected),
-      selectionListener: this,
     );
+  }
 
-    // Compute padding around chart.
-    _chartController.setViewPortOffsets(
-        defaultSpacing * 3, denseSpacing, defaultSpacing, defaultSpacing);
+  Widget _buildFrame(TimelineFrame frame) {
+    return FlutterFramesChartItem(
+      frame: frame,
+      selected: frame == _selectedFrame,
+      onSelected: () => _controller.selectFrame(frame),
+      msPerPx: msPerPx,
+      availableChartHeight: availableChartHeight,
+      displayRefreshRate: widget.displayRefreshRate,
+    );
   }
 
-  void onBarSelected(int index) {
-    _controller.selectFrame(_controller.data.frames[index + indexOffset]);
+  Widget _buildChartLegend() {
+    return Column(
+      key: FlutterFramesChart.chartLegendKey,
+      mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        _legendItem('Frame Time (UI)', mainUiColor),
+        _legendItem('Frame Time (Raster)', mainRasterColor),
+        _legendItem('Jank (slow frame)', uiJankColor),
+      ],
+    );
   }
 
-  void _initData() {
-    // Create place holder for empty chart.
-    // TODO(terry): Look at fixing MPFlutterChart to handle empty data entries.
-    _frameDurations.add(createStubBarEntry());
+  Widget _legendItem(String description, Color color) {
+    return Row(
+      children: [
+        Container(
+          height: legendSquareSize,
+          width: legendSquareSize,
+          color: color,
+        ),
+        const SizedBox(width: denseSpacing),
+        Text(description),
+      ],
+    );
+  }
+}
 
-    // Create heap used dataset.
-    frameDurationsSet = BarDataSet(_frameDurations, 'Durations')
-      ..setColors1([mainRasterColor, mainUiColor])
-      ..setDrawValues(false);
+class FlutterFramesChartItem extends StatelessWidget {
+  const FlutterFramesChartItem({
+    @required this.frame,
+    @required this.selected,
+    @required this.onSelected,
+    @required this.msPerPx,
+    @required this.availableChartHeight,
+    @required this.displayRefreshRate,
+  });
 
-    // Create a data object with all the data sets - stacked bar.
-    _chartController.data = BarData([]..add(frameDurationsSet));
+  static const defaultFrameWidth = 32.0;
```

Comment:
Should a TODO be added to compute this based on pixel density and device (32 might be too small for higher density screens).  I wouldn't fix just a note.

Reply:
This is the legend size? not bar width just noticed that below.  I guess that should be computed we have this kind of thing everywhere in devtools.